diff --git a/bazel/protobuf.bzl b/bazel/protobuf.bzl
--- a/bazel/protobuf.bzl
+++ b/bazel/protobuf.bzl
@@ -1,3 +1,4 @@
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
 load("@bazel_tools//tools/jdk:toolchain_utils.bzl", "find_java_runtime_toolchain", "find_java_toolchain")
 load("@rules_proto//proto:defs.bzl", "ProtoInfo")
